### PR TITLE
Feat/validate

### PR DESCRIPTION
**Pull request by giangbui**
Given a manifest, check if all the objects exist in cloud resources and are indexed in indexd or not. The output is argumented manifest with `aws url`, `gs url` and `indexd urls`

### CODE REVIEW

File: scripts/validate.py [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+                object_path = "{}/{}".format(
+                    gs_bucket, utils.get_structured_object_key(fi["id"], ignored_dict)
+                )
+            else:
+                object_path = "{}/{}/{}".format(gs_bucket, fi["id"], fi["file_name"])
+
+            if object_path not in gs_copied_objects and fi["size"] != 0:
+                total_gs_copy_failures += 1
+                logger.error(
+                    "{} is not copied yet to google buckes".format(object_path)
```
Comment (paulineribeyre) [inline scripts/validate.py]:
```suggestion
                    "{} is not copied yet to google buckets".format(object_path)
```

File: scripts/validate.py [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+                if fi["aws_url"] not in fi["indexd_url"]:
+                    total_aws_index_failures += 1
+                    logger.error("indexd does not have aws url of {}".format(fi["id"]))
+
+            # validate google
+            gs_bucket = utils.get_google_bucket_name(fi, PROJECT_ACL)
+            if fi["id"] in ignored_dict:
+                object_path = "{}/{}".format(
+                    gs_bucket, utils.get_structured_object_key(fi["id"], ignored_dict)
+                )
```
Comment (paulineribeyre) [inline scripts/validate.py]: maybe add in a comment what you told me? `if a file is in ignored_dict, we don’t copy but we still need to check if the file exists in the bucket and is indexed`

File: scripts/utils.py [hunk 3]
```diff
@@ -266,3 +266,23 @@ def build_source_bucket_dataset(bucket_name, objects):
         th.join()
 
     return copied_object
+
+
+def write_csv(filename, files, sorted_attr=None, fieldnames=None):
+    def on_key(element):
+        return element[sorted_attr]
+    import csv
```
[the remark this answers is not in the document]
Comment (giangbui) [reply scripts/utils.py]: Thanks! my bad copy

File: scripts/validate.py [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+        total_gs_copy_failures = 0
+        total_aws_index_failures = 0
+        total_gs_index_failures = 0
+        manifest_file = manifest_file.strip()
+        files = utils.get_fileinfo_list_from_s3_manifest(manifest_file)
+        for fi in files:
+            del fi["url"]
+            fi["aws_url"], fi["gs_url"], fi["indexd_url"] = None, None, None
+
+            fi["indexd_url"] = indexd_records.get(fi.get("id"))
```
[the remark this answers is not in the document]
Comment (giangbui) [reply scripts/validate.py]: Actually loading all the records to dictionary makes the process much faster

File: scripts/validate.py (hunk 1 above)
Comment (giangbui) [reply scripts/validate.py]: Thanks!

**Pull request merged**